sourcetracker/_plot.py

```diff
@@ -10,16 +10,306 @@
+            tly = ax.get_yticklabels()
+            ax.set_yticklabels(tly, rotation=0)
+        if normalized:
+            if unknowns:
+                add_line = tra + "_pairedheatmap_normalized.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.out_name + add_line))
+            else:
+                add_line = tra + "_pairedheatmap_nounknown_normalized.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.out_name + add_line))
+        else:
+            if unknowns:
+                add_line = tra + "_pairedheatmap.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.out_name + add_line))
+            else:
+                add_line = tra + "_pairedheatmap_nounknowns.png"
+                plt.savefig(os.path.join(self.file,
+                                         self.out_name + add_line))
+
+    def ST_Stacked_bar(self, unknowns=True, x_lab="Sink",
+                       y_lab="Source Proportion", coloring=[], flipped=False):
+        """
+        Creates a Stacked bar plot for the user with direct png save function
+        Parameters
+        ----------
+        unknowns bool
+            removes unknown column
+        xlabel string
+            x label
+        ylabel string
+            y label
+        coloring string list
+            string list of colors to encode the bar plot.
+            must have and equal number of colors as sources
+        flipped bool
+            flips x and y axis
+        returns
+        --------
+        none
+        outputs a stacked bar visualization in a PNG
+
+        color example list
+
+        '#1f77b4'Blue, '#ff7f0e'Orange, '#2ca02c'Green, '#d62728'Red,
+        '#9467bd'Purple, '#8c564b'Brown, '#e377c2'Pink, '#7f7f7f'Grey,
+        '#bcbd22'Gold, '#17becf'Cyan
+        make sure to use contrasting colors in order better illuminate
+        your data above are some example codes to use
+        """
+        prop = self.mpm
+        if not unknowns:
+            prop = prop.drop(['Unknown'], axis=1)
+            prop = prop.div(prop.sum(axis=1), axis=0)
+        if flipped:
+            prop = prop.T
+            y_lab_flip = x_lab
+            x_lab_flip = y_lab
+            y_lab = y_lab_flip
+            x_lab = x_lab_flip
+            prop = prop.div(prop.sum(axis=1), axis=0)
+        prop = prop.reset_index()
+        if len(coloring) != (prop.shape[1]-1):
+            coloring = []
+        if coloring != []:
+            prop.plot(kind='bar', x=prop.columns[0], stacked=True,
+                      figsize=((prop.shape[1] * 3 / 4)+4,
+                      (prop.shape[0] * 3 / 4)+4),
+                      color=coloring)
+        else:
+            prop.plot(kind='bar', x=prop.columns[0], stacked=True,
+                      figsize=((prop.shape[1] * 3 / 4)+4,
+                      (prop.shape[0] * 3 / 4)+4))
```

Comment:
```suggestion
        if coloring = []:
            coloring = None

       prop.plot(kind='bar', x=prop.columns[0], stacked=True,
                     figsize=((prop.shape[1] * 3 / 4)+4,
                     (prop.shape[0] * 3 / 4)+4),
                     color=coloring)
```

Reply:
Would this work? I never wanted to open that can of worms honestly.

Reply:
Looking at the matplotlib backend documentation, this should work. 

Please test this suggestion and modify if needed.